packages/bits/src/lib/wizard-v2/wizard-icon.directive.ts

```diff
@@ -0,0 +1,26 @@
+import {Directive, Input, TemplateRef} from "@angular/core";
+import {StepState} from "@angular/cdk/stepper";
+
+/** Template context available to an attached `NuiWizardIcon`. */
+export interface NuiWizardIconContext {
+    index: number;
+    active: boolean;
+    optional: boolean;
+}
+
+export interface MatStepperIconContext {
+    index: number;
+    active: boolean;
+    optional: boolean;
+}
+
+@Directive({
+    selector: "ng-template[nuiWizardIcon]",
+})
+export class NuiWizardIconDirective {
```

Comment:
I wonder if a better name might be `WizardStepStateDirective`

Reply:
@Blakesalot I think WizardIconDirective is a better name for it because this directive is used only for overriding icons.